--- start.before.sh	2026-09-11 13:39:52.810349823 +0200
+++ start.after.sh	2026-09-11 13:40:26.657894003 +0200
@@ -1,8 +1,11 @@
 #!/bin/bash
 
 echo "Imaginary has started"
-if [ -z "$IMAGINARY_SECRET" ]; then
-    imaginary -return-size -max-allowed-resolution 222.2 "$@"
-else
-    imaginary -return-size -max-allowed-resolution 222.2 -key "$IMAGINARY_SECRET" "$@"
-fi
\ No newline at end of file
+
+IMAGINARY_ARGS=(-return-size -max-allowed-resolution 222.2)
+
+if [ -n "$IMAGINARY_SECRET" ]; then
+    IMAGINARY_ARGS+=(-key "$IMAGINARY_SECRET")
+fi
+
+exec imaginary "${IMAGINARY_ARGS[@]}" "$@"
\ No newline at end of file
