--- start.before.sh	2026-09-11 15:25:56.470941337 +0200
+++ start.after.sh	2026-09-11 15:26:44.415540146 +0200
@@ -1,8 +1,11 @@
 #!/bin/bash
 
 echo "Imaginary has started"
-if [ -z "$IMAGINARY_SECRET" ]; then
-    imaginary -return-size -max-allowed-resolution 222.2 "$@"
-else
-    imaginary -return-size -max-allowed-resolution 222.2 -key "$IMAGINARY_SECRET" "$@"
+
+IMAGINARY_ARGS=(-return-size -max-allowed-resolution 222.2)
+
+if [ -n "$IMAGINARY_SECRET" ]; then
+    IMAGINARY_ARGS+=(-key "$IMAGINARY_SECRET")
 fi
+
+exec imaginary "${IMAGINARY_ARGS[@]}" "$@"
